homeassistant/components/proxmoxve/binary_sensor.py

```diff
@@ -1,112 +1,95 @@
+                vm_name = coordinator_data["name"]
+                vm_status = create_binary_sensor(
+                    coordinator, host_name, node_name, vm_id, vm_name
+                )
+                sensors.append(vm_status)
 
-    def __init__(self, proxmox_client, item_node, item_type, item_id):
-        """Initialize the binary sensor."""
-        self._proxmox_client = proxmox_client
-        self._item_node = item_node
-        self._item_type = item_type
-        self._item_id = item_id
+            for container_id in node_config["containers"]:
+                coordinator_data = coordinator.data[host_name][node_name][container_id]
 
-        self._vmname = None
-        self._name = None
+                # unfound container case
+                if coordinator_data is None:
+                    continue
 
-        self._state = None
+                container_name = coordinator_data["name"]
+                container_status = create_binary_sensor(
+                    coordinator, host_name, node_name, container_id, container_name
+                )
+                sensors.append(container_status)
 
-    @property
-    def name(self):
-        """Return the name of the entity."""
-        return self._name
+    add_entities(sensors)
 
-    @property
-    def is_on(self):
-        """Return true if VM/container is running."""
-        return self._state
 
-    @property
-    def device_state_attributes(self):
-        """Return device attributes of the entity."""
-        return {
-            "node": self._item_node,
-            "vmid": self._item_id,
-            "vmname": self._vmname,
-            "type": self._item_type.name,
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-        }
-
-    def update(self):
-        """Check if the VM/Container is running."""
-        item = self.poll_item()
-
-        if item is None:
-            _LOGGER.warning("Failed to poll VM/container %s", self._item_id)
-            return
-
-        self._state = item["status"] == "running"
-
-    def poll_item(self):
-        """Find the VM/Container with the set item_id."""
-        items = (
-            self._proxmox_client.get_api_client()
-            .nodes(self._item_node)
-            .get(self._item_type.name)
-        )
-        item = next(
-            (item for item in items if item["vmid"] == str(self._item_id)), None
-        )
+def create_binary_sensor(coordinator, host_name, node_name, vm_id, name):
+    """Create a binary sensor based on the given data."""
+    return ProxmoxBinarySensor(
+        coordinator=coordinator,
+        unique_id=f"proxmox_{node_name}_{vm_id}_running",
+        name=f"{node_name}_{name}_running",
+        icon="",
+        host_name=host_name,
+        node_name=node_name,
+        vm_id=vm_id,
+    )
+
 
-        if item is None:
-            _LOGGER.warning("Couldn't find VM/Container with the ID %s", self._item_id)
-            return None
+class ProxmoxBinarySensor(ProxmoxEntity):
```

Comment:
Binary sensor entities should inherit `BinarySensorEntity`. Please add that class to the inheritence.